delivery-receipts-api/src/main/java/uk/gov/di/authentication/deliveryreceiptsapi/lambda/NotifyCallbackHandler.java

```diff
@@ -0,0 +1,116 @@
+        LOG.info("Received request");
+        validateBearerToken(input.getHeaders());
+        NotifyDeliveryReceipt deliveryReceipt;
+        try {
+            deliveryReceipt =
+                    new ObjectMapper().readValue(input.getBody(), NotifyDeliveryReceipt.class);
+            if (deliveryReceipt.getNotification_type().equals("sms")) {
+                var countryCode = getCountryCodeFromNumber(deliveryReceipt.getTo());
+                var deliveryStatus = getDeliveryStatus(deliveryReceipt.getStatus());
+                LOG.info(
+                        "SmsDeliveryStatus: {}, NotifyStatus: {}, CountryCode: {}",
+                        deliveryStatus,
+                        deliveryReceipt.getStatus(),
+                        countryCode);
+                cloudwatchMetricsService.incrementCounter(
+                        deliveryStatus,
+                        Map.of(
+                                "CountryCode",
+                                String.valueOf(countryCode),
+                                "Environment",
+                                configurationService.getEnvironment(),
+                                "NotifyStatus",
+                                deliveryReceipt.getStatus()));
+                LOG.info("SMS callback request processed");
+            }
+        } catch (JsonProcessingException e) {
+            LOG.error("Unable to parse Notify Delivery Receipt");
+            throw new RuntimeException("Unable to parse Notify Delivery Receipt");
+        }
+        return null;
+    }
+
+    private void validateBearerToken(Map<String, String> headers) {
+        var notifyCallbackBearerToken = configurationService.getNotifyCallbackBearerToken();
+        if (Objects.isNull(headers.get("Authorization"))
+                || !headers.get("Authorization").startsWith("Bearer ")) {
+            LOG.error("No bearer token in request");
+            throw new RuntimeException("No bearer token in request");
+        }
+        var token = headers.get("Authorization").substring(7);
+        if (!token.equals(notifyCallbackBearerToken)) {
+            LOG.error("Invalid bearer token in request");
+            throw new RuntimeException("Invalid bearer token in request");
+        }
+    }
+
+    private int getCountryCodeFromNumber(String number) {
+        String defaultRegion = null;
+        if (!number.startsWith("+")) {
+            defaultRegion = "GB";
+        }
+        var phoneUtil = PhoneNumberUtil.getInstance();
+        try {
+            return phoneUtil.parse(number, defaultRegion).getCountryCode();
+        } catch (NumberParseException e) {
+            LOG.error("Unable to parse number");
+            throw new RuntimeException("Unable to parse number");
+        }
+    }
+
+    private String getDeliveryStatus(String notifyStatus) {
+        var deliveryStatus = SMS_FAILURE.toString();
+        if (notifyStatus.equals("delivered")) {
```

Comment:
```suggestion
        if ("delivered".equals(notifyStatus) {
```
If `notifyStatus` is null, it would cause a NullPointerException, reversing the check order removes that risk

Reply:
Good shout